R/nloptr.R

```diff
@@ -313,16 +313,22 @@ function( x0,
             if( any(is.na(m1)) ){
                 mx1 = which( is.na(m1) )
                 for( i in 1:length(mx1) ){
-                    stop(paste(funname, " requires argument '", fnms[mx1[i]], "' but this has not been passed to the 'nloptr' function.\n", sep = ""))
+                    has_default = !(fnms[mx1[i]] == "")
```

Comment:
You should be checking the value in the list here, not the name, i.e. `has_default = !(flist[[mx1[i] + 1]] == "")`.